File: flask-backend/controllers/postController.py
from MySQLdb.cursors import DictCursor

def get_all_posts(mysql):
    cursor = mysql.connection.cursor(DictCursor)
    cursor.execute("SELECT * FROM posts;")  # Replace with your table name
    posts = cursor.fetchall()
    cursor.close()
    return posts

def create_post(mysql, post_data):
    title = post_data.get('title')
    date = post_data.get('date')
    label = post_data.get('label')
    description = post_data.get('description')
    if not title or not date or not label or not description:
        return {"error": "Title, date, label and description are required"}, 400

    cursor = mysql.connection.cursor()
    cursor.execute('INSERT INTO posts (post_title, post_date, post_label, post_description) VALUES (%s, %s, %s, %s);', (title, date, label, description))
    mysql.connection.commit()
    cursor.close()

    return {"message": "Post created successfully"}, 201

File: express-backend/controllers/postController.js
const db = require('../models/db');

// Get all users
const getAllPosts = async (req, res) => {
    try {
        const posts = await db.query('SELECT * FROM posts;');
        res.status(200).json(posts);
    } catch (error) {
        res.status(500).json({ error: 'Failed to fetch posts' });
    }
};

// Create a new user
const createPost = async (req, res) => {
    const { title, date, label, description } = req.body;
    try {
        const result = await db.query('INSERT INTO posts (post_title, post_date, post_label, post_description) VALUES (?, ?, ?, ?);', [title, date, label, description]);
        res.status(201).json({ message: 'Post created', postId: result.insertId });
    } catch (error) {
        res.status(500).json({ error: 'Failed to create post' });
    }
};

module.exports = { getAllPosts, createPost };